README.md

```diff
@@ -179,10 +179,100 @@ If you're calling a **GET** on `/entities` , you're asking for a list of entitie
+
+> Creating a user:
+
+- **POST** `/users`
+  This creates a user with the info you pass along in the request body.
+
+> Deleting a post
+
+- **DELETE** `/users/:id`
+  This Deletes the user with the id you pass along in the url
+
+###### Bad
+
+> Getting a post
+
+- **GET** `/get/posts?id=:id`
+  This is wrong because you already established that it is a GET. Using /get on the endpoint name is redundant
+
+> Creating a post
+
+- **POST** `/create/post`
+  Same as above, you already estabished that you're going to create something with the **POST** so, again, the create is redundant
+
+> Deleting a post
+
+- **POST** `/delete_post/:id`
+  In thi case not only you're being redundant by putting the verb delete on the endpoint name, but you're doing a wrong redundancy because it is actually a **POST** request and they're not for deleting objects
+##### Reasoning
+
+Your endpoind should reflect what they do based on the HTTP Verb plus the endpoing name (noun). So it make sense that doing a **GET** to `/entities` retrieves a list of entities. In exchange, doing a **GET** to `/get/entity` looks redundant and is not self descriptive. One more example: **POST** to `/update_entity/:id` that, semantically speaking, means you're Creating an `update_entity` object with the id on the url IS WRONG. This should be achived by **PUT** or **PATCH** `/entities/:id`
+
+---
+
+#### Plural nouns or singular nouns?
+
+> If your endpoint retrieves a list or a group of entities instead of just one then it should be named with a plural noun. If it always contains a single entity then it should be named with a singular noun.
+
+##### Examples
+
+###### Good
+
+> Getting comments from a post
+
+- **GET** `/posts/:id/comments`
+  The endpoint retrieves a list of comments for the post with the id you pass along in the url.
+
+> Getting the likes count of a post
+
+- **GET** `/posts/:id/like_count`
+  The endpoint  the likes count of  a post with the id you pass along in the url.
+
+> Deleting all the comments from a post
+
+- **DELETE** `/posts/:id/comments`
+  The endpoint deletes all the comments from the post with the id you pass along in the url
+
+###### Bad
+
+> Getting a post
+
+- **GET** `/post/:id`
+  You're getting a post from a list/group of posts, so it should be plural `/posts/:id`
+
+> Creating a post
+
+- **PATCH** `/posts/`
+  You're trying to update a post but you're not passing the id on the url. Passing it along in the request body is wrong.
+
+> Deleting a comment from a post
+
+- **DELETE** `/posts/:post_id/comment/:id`
+  Here you have a list/group of comments so it should be plural `/posts/:post_id/comments/:id`. I changed `comment` for `comments`
+
+##### Reasoning
+
+Your endpoint name should be clear about the fact that it affects a single object or a group/list of objects. When you have singular resources, use singular nouns. When you have multiple resources affected by and endpoint, use plural nouns.
```

Comment:
No instructions in reasoning. Only reasons.
And please check the rest of the file with that in mind.

Also add something stating how it's _coherent_ to see urls like `/posts` and `posts/:id` and it's clear in that way that the post you create with `POST /posts` ends up in `GET /posts/:id`.